From c8eef724dbfbe19391a6acce58b290544fd4653b Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Mon, 18 Apr 2011 22:09:46 -0400
Subject: [PATCH 232/248] Fix disabled channel timer start/stop handling

Should test if channels are active before calling timer start/stop,
otherwise causes NULL pointer exception.

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 ltt/ltt-tracer.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/ltt/ltt-tracer.c b/ltt/ltt-tracer.c
index 24a3a92..1ca5ba2 100644
--- a/ltt/ltt-tracer.c
+++ b/ltt/ltt-tracer.c
@@ -1031,7 +1031,9 @@ void ltt_channels_trace_start_timer(struct ltt_chan *channels,
 
 	for (i = 0; i < nr_channels; i++) {
 		struct ltt_chan *chan = &channels[i];
-		chan->a.trace->ops->start_switch_timer(chan);
+
+		if (chan->active)
+			chan->a.trace->ops->start_switch_timer(chan);
 	}
 }
 
@@ -1046,7 +1048,9 @@ void ltt_channels_trace_stop_timer(struct ltt_chan *channels,
 
 	for (i = 0; i < nr_channels; i++) {
 		struct ltt_chan *chan = &channels[i];
-		chan->a.trace->ops->stop_switch_timer(chan);
+
+		if (chan->active)
+			chan->a.trace->ops->stop_switch_timer(chan);
 	}
 }
 
-- 
1.7.0.4

